service/history/taskProcessor.go

```diff
@@ -246,9 +249,20 @@ func (t *taskProcessor) processTaskOnce(
 	default:
 	}
 
+	var scopeIdx int
+	var err error
+
+	domainID := task.task.GetDomainID()
+	scope := t.metricsScopeCache.Get(domainID)
```

Comment:
We have different scopes for the different task types, so `domainID` alone can't uniquely identify a metric scope.